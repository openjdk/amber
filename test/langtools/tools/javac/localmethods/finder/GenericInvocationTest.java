/**
 * @test /nodynamiccopyright/
 * @bug 8233754
 * @compile/fail/ref=GenericInvocationTest.out -Werror -XDrawDiagnostics -XDfind=local-methods GenericInvocationTest.java
 */

public class GenericInvocationTest {

    private static <T> T id(T t ) {
        return t;
    } // can be local

    private static <T> T id(T t1, T t2) {
        return null;
    }  // can be local.

    private static <T> T id(T ... t) {
        return null;
    }  // cannot be local since is invoked with explicit type witness.

    public static void main(String [] args) {
        GenericInvocationTest.<String> id (id(id("hello "), id("world ")), "is ", "the ", "message!");
    }
}
