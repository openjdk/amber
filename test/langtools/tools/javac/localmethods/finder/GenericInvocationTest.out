GenericInvocationTest.java:9:26: compiler.warn.method.could.be.declared.local: main(java.lang.String[]), kindname.class, GenericInvocationTest
GenericInvocationTest.java:13:26: compiler.warn.method.could.be.declared.local: main(java.lang.String[]), kindname.class, GenericInvocationTest
- compiler.err.warnings.and.werror
- compiler.note.unchecked.filename: GenericInvocationTest.java
- compiler.note.unchecked.recompile
1 error
2 warnings
